src/gc/gc.cpp

```diff
@@ -10602,7 +10602,18 @@ gc_heap::init_gc_heap (int  h_number)
     make_background_mark_stack (b_arr);
 #endif //BACKGROUND_GC
 
-    adjust_ephemeral_limits();
+    ephemeral_low = generation_allocation_start(generation_of(max_generation - 1));
+    ephemeral_high = heap_segment_reserved(ephemeral_heap_segment);
```

Comment:
Since those fileds are `PER_HEAP` we have to update them on each invocation. That was the bug.

Reply:
was this not the case with `adjust_ephemeral_limits`?

Reply:
not sure what  you mean? adjust_ephemeral_limits was updating the per heap ephemeral_low/high. when I wrote that code example for the change, I mistakenly thought @Rattenkrieg was just using some local vars called ephemeral_low/ephemeral_high so I was only updating them when heap_number is 0.

Reply:
ah, I see what happened here; sorry, I understand now (I missed the previous change to this code)